services/viva-ms/lambdas/syncWorkflowAndSetStatus.js

```diff
@@ -13,9 +13,31 @@ export async function main(event) {
   const personalNumber = event.detail.user.personalNumber;
   const PK = `USER#${personalNumber}`;
 
-  const allUserCases = await getAllUserCases(PK);
+  const [getAllUserCasesError, allUserCases] = await to(getAllUserCases(PK));
+  if (getAllUserCasesError) {
+    return console.error('(Viva-ms) DynamoDB query failed', getAllUserCasesError);
+  }
+
+  const userCaseItems = allUserCases.Items;
+  if (userCaseItems === undefined || userCaseItems.length === 0) {
+    return console.error('(Viva-ms) DynamoDB query did not fetch any cases');
+  }
+
+  for (const userCase of userCaseItems) {
+    const workflowId = userCase.details.workflowId;
+
+    const [myPagesError, myPagesResponse] = await to(
+      vivaAdapter.workflow.get({ personalNumber, workflowId })
```

Comment:
What we could do here to reduce request time is to get all workflows, since the adapter does the same request either way but have a filter on the /workflow/{id} to only return the requested workflow. In other words both /workflows and /workflows/{id} sends the same request to viva from vada

Reply:
Might be a broader thing to discuss.

Reply:
Might be a cause for improved requests towards viva in in the viva adapter